### DISCUSSION TITLE
Screen Name

### DISCUSSION THREAD

**Discussion by 3DGera** [pixelit-project/PixelIt#363] [General]
Hallo, ich habe von Awtrix zu euch gewechselt =) nun habe ich aber ein Problem mit meinen Automationen.

Die Awtrix hat im MQTT den Screen Namen mit zurückgegeben und auf diesen habe ich dann meinen Anzeigen aktualisiert. Leider gibt PixlIt keinen Screen Namen zurück. Ist es möglich diesen mit einzubauen?

**Comment by foorschtbar** [comment]
Anders. Du schiebst die Screens (Daten) in den Screen Update Node und setzt ne Playlist mit den Screen Names und der Core Node kümmert sich drum das immer der aktuellste Screen angezeigt wird. Mir fällt nicht ein wofür du wissen musst was gerade angezeigt wird, und wenn bekommst du das ja aus dem MQTT Topic das der Core Node benutzt

**Comment by 3DGera** [comment]
Da reden wir wohl gerade aneinander vorbei

Ich schicke initial einen Screen, wenn die Anzeige z.B. neu startet. Danach läuft dieser Screen ja im Wechsel mit den anderen, das ist soweit klar. Aber in meinem Screen ändern sich ja ständig die Werte und deshalb muss ich den Screen aktualisieren. Das mache ich mit dem Screen-Namen. Sobald dieser aufgerufen wird, wird der Screen mit den aktuellen Werten neu beschrieben.

Hier mal meine Screen-Payload:

`{
    "text": {
        "textString": "{{(states('sensor.electricity_gesamtleistung')|float(default=0))}}W Solar: {{(states('sensor.pvanlage_energy_power')|float(default=0) + states('sensor.growatt_growatt_inverter_dcpower')|float(default=0))}}W",
        "scrollText": "auto",
        "bigFont": false,
        "centerText": true,
        "scrollTextDelay": 40,
        "hexColor": "#FFFFFF",
        "position": {
        "x": 0,
        "y": 1
        }
    }
}`

Die Sensoren _sensor.electricity_gesamtleistung_, _sensor.pvanlage_energy_power_, _sensor.growatt_growatt_inverter_dcpower_ ändern ja ständig ihre Werte. Wie aktualisiere ich diese dann?

**Comment by foorschtbar** [comment]
Ahhh, du setzt NICHT Node-RED ein, da läuft das wie beschrieben von mir. Wie das bei HA (oder was benutzt du?) funktioniert, weiß ich nicht. Vielleicht mal in der Telegramgruppe fragen.

**Comment by 3DGera** [comment]
Genau, ich nutze Home Assistant als Zentrale und möchte die PixelIT mittels MQTT ansteuern. Das funktioniert auch, bloß wechselt er dann nicht richtig durch die Screens und aktualisiert diese auch nicht, da meine Automation nicht weiß, in welchem Screen die PixelIT ist.

Man könnte zwar alle Screens per "Alert Screen" laufen lassen, das ist aber nicht Sinn der Sache, oder?

Daher wäre es cool, wenn die PixelIT per MQTT mitteilen würde, in welchem Screen sie gerade ist. So könnte man in der Automation sagen: Wenn zu Screen X gewechselt wird, schicke Screen X die neuen Werte. So habe ich es bei Awtrix gemacht.

Ich habe mich zwar am Wochenende mal ein wenig mit Node-RED beschäftigt, und ja, das geht auch klasse, aber wofür ist dann noch die MQTT-Schnittstelle gut, wenn man es doch über Node-RED machen müsste? =)

**Comment by foorschtbar** [comment]
Du suchst da den "Fehler" an der falschen Seite. Die PixelIt selber ist "dumm" und bekommt ihre Inhalte (Screens) von extern. Wieso sollte sie informieren, wenn ihr ein neuer Screen anzeigt? Sie kann nur den einen Screen anzeigen der ihr gerade geschickt wurde. Wenn du für deine Automatisierung diese Info brachst, subscripe auf das Topic und du hast die Info. 

Wie gesagt, frag vielleicht mal im Chat wie andere das mit HA gelöst haben.

Node-RED kann über die PixelIt Nodes sowohl per REST als auch per MQTT mit einer oder mehreren PixelIts sprechen. Letzteres ist der vorteil von MQTT. Man kann mehrere PixelIts sycron halten mit minmalen Aufwand.

**Comment by 3DGera** [comment]
Ich möchte niemandem einen Fehler zuschieben!

Wie du vielleicht schon mitbekommen hast, habe ich vorher Awtrix genutzt. Da dieses Projekt aber schon sehr lange nicht mehr supportet wird, habe ich diese Alternative gefunden, da beide Projekte im Grunde genommen das Gleiche machen.

In Node-RED kann man Playlisten vorgeben, diese müssten dann in PixelIT irgendwie verarbeitet werden, oder nicht? Dann könnte man auch zurückgeben, welcher Screen gerade zu sehen ist. Aber gut, ich will nicht weiter diskutieren.

Ich werde das Ganze dann wohl doch per Node-RED machen. ;)
